packages/yoastseo/src/parse/build/private/filterBeforeTokenizing.js

```diff
@@ -0,0 +1,36 @@
+import { isEmpty } from "lodash-es";
+
+/**
+ * Removes the 'text' child nodes from 'code' and 'script' nodes, and the startTag and endTag properties of the
+ * sourceCodeLocation of 'code' and 'script' nodes.
+ *
+ * The text child nodes are removed because we don't want to include text between 'code' and 'script' nodes in the analysis.
+ * If we don't remove them here, the text between 'code' and 'script' nodes will be considered as part of the paragraph's
+ * inner text and will be tokenized.
+ *
+ * The startTag and endTag properties are removed so that when sentence/token positions are calculated, the whole length
+ * of the code/script element is considered when calculating the positions - not just the length of the tags.
+ * For most other child nodes, we only need to add the length of the child nodes tags to the sentence/token positions
+ * because what's in between those tags is part of the sentence/token itself. But in case of nodes like 'code' and 'script'
+ * where the content is not part of the sentence/token, we need to add the length of the whole element to the
+ * sentence/token positions. The way that the current algorithm for calculating positions works is that only if the child
+ * node doesn't have startTag and endTag properties, the length of the whole element instead of just the tags is considered.
+ *
+ * @param {Node} node The node to check.
+ *
+ * @returns {Node} The filtered out node that's ready to be tokenized.
+ */
+export function filterBeforeTokenizing( node ) {
+	if ( node.name === "code" || node.name === "script" ) {
+		node.childNodes = node.childNodes.filter( childNode => childNode.name !== "#text" );
+		delete node.sourceCodeLocation.startTag;
+		delete node.sourceCodeLocation.endTag;
+	}
+
+	// Recursively filters the node's children.
+	if ( ! isEmpty( node.childNodes ) ) {
+		node.childNodes.map( filterBeforeTokenizing );
+	}
+
+	return node;
```

Comment:
Currently, this approach doesn't strip the 'text' child nodes from 'code'/'script' child nodes. For example for this HTML:
```
const html = "<p>Some text and code <code><strong>console.log</strong>( code )</code></p>";
```
In above HTML, we would have the following result:
```
"childNodes": [
        {"name": "#text", "value": "Some text and code "},
        {
          "attributes": {},
          "childNodes": [
            {
              "attributes": {},
              "childNodes": [
                {"name": "#text", "value": "console.log"}
              ],
              "name": "strong",
              "sourceCodeLocation": {
                "endOffset": 56,
                "endTag": {"endOffset": 56, "startOffset": 47},
                "startOffset": 28,
                "startTag": {"endOffset": 36, "startOffset": 28}
              }
            }
          ],
          "name": "code",
          "sourceCodeLocation": {
            "endOffset": 71,
            "startOffset": 22
          }
        }
      ],
```
I assume that we would also want to strip the 'text' of the 'strong' element inside 'code' node?

Reply:
Somehow the recursion doesn't work on nested code/script elements